test/functional/fixtures/run-options/allow-multiple-windows/test.js

```diff
@@ -46,4 +48,22 @@ describe('Allow multiple windows', () => {
     it('Close the window immediately after opening (GH-3762)', () => {
         return runTests('testcafe-fixtures/close-window-immediately-after-opeping.js', null, { allowMultipleWindows: true });
     });
+
+    it('headless', () => {
+        return createTestCafe('127.0.0.1', 1335, 1336)
+            .then(tc => {
+                testCafe = tc;
+            })
+            .then(() => {
+                const fullTestPath = path.join(__dirname, './testcafe-fixtures/headless.js');
+
+                return testCafe.createRunner()
+                    .browsers(`chrome:headless`)
+                    .src(fullTestPath)
+                    .run({ allowMultipleWindows: true });
+            })
+            .then(() => {
+                return testCafe.close();
+            });
+    });
```

Comment:
How can we be sure that the test actually runs in multiple windows?

Reply:
[This](https://github.com/DevExpress/testcafe/blob/master/test/functional/fixtures/run-options/allow-multiple-windows/test.js#L4) group of tests checks such condition.

Algorithm: 
* check the current page title
* switch to child window
* check page title of the opened window